src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -139,6 +139,7 @@ boolean canAccessRepository(Long participationId) throws IllegalArgumentExceptio
     @Override
     @GetMapping(value = "/repository/{participationId}/files", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<Map<String, FileType>> getFiles(@PathVariable Long participationId) {
+        // TODO: we need to fetch the commit which belongs to the last not ILLEGAL submission!
```

Comment:
What about this todo? Will this be done in a followup?

Reply:
This might be implemented in the future since it's a bit more complicated. I removed the todos for now.